### PR TITLE
Add proposed name to pages

### PR DESCRIPTION
**Pull request by HatiDev**
### JIRA link

https://companieshouse.atlassian.net/browse/LP-316

### Change description

Add proposed name to pages

### Work checklist

- [ ] Tests added where applicable
- [ ] UI changes look good on mobile
- [x] UI changes meet accessibility criteria

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: src/views/email.njk [hunk 1]
```diff
@@ -6,9 +6,10 @@
   <div class="govuk-grid-row">
     <div class="govuk-grid-column-two-thirds">
     <form class="form" action={{ props.currentUrl }} method="post">
-      <input type="hidden" name="pageType" value={{ props.pageType }}>
-      
+      <input type="hidden" name="pageType" value={{ props.pageType }}>      
       {% include "includes/csrf_token.njk" %}
+      
+      {% include "includes/proposed-name.njk" %}
```
Comment (lduranteau) [inline src/views/email.njk:12]: Can you update the tests to check that it is really present on the page?

File: src/views/general-partner-choice.njk [hunk 2]
```diff
@@ -7,9 +7,10 @@
     <div class="govuk-grid-column-two-thirds">
       <form class="form" action={{ props.currentUrl }} method="post">
         <input type="hidden" name="pageType" value={{ props.pageType }}>
-
         {% include "includes/csrf_token.njk" %}
         
+        {% include "includes/proposed-name.njk" %}
```
Comment (lduranteau) [inline src/views/general-partner-choice.njk:12]: same as above

File: src/views/general-partners.njk [hunk 3]
```diff
@@ -5,8 +5,10 @@
 {% block content %}
   <div class="govuk-grid-row">
     <div class="govuk-grid-column-two-thirds">
- 
-        <h1 class="govuk-heading-xl">{{ i18n.generalPartnersPage.title }}</h1>
+
+      {% include "includes/proposed-name.njk" %}
```
Comment (lduranteau) [inline src/views/general-partners.njk:9]: same as above

File: src/views/limited-partner-choice.njk [hunk 4]
```diff
@@ -9,6 +9,8 @@
       <form class="form" action="{{ props.currentUrl }}" method="post">
         <input type="hidden" name="pageType" value={{ props.pageType }}>
         {% include "includes/csrf_token.njk" %}
+
+        {% include "includes/proposed-name.njk" %}
```
Comment (lduranteau) [inline src/views/limited-partner-choice.njk:13]: same as above

File: src/views/limited-partners.njk [hunk 5]
```diff
@@ -5,6 +5,8 @@
 {% block content %}
   <div class="govuk-grid-row">
     <div class="govuk-grid-column-two-thirds">
+
+        {% include "includes/proposed-name.njk" %}
```
Comment (lduranteau) [inline src/views/limited-partners.njk:9]: same as above

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

File: src/presentation/test/integration/registration/general-partner-choice.test.ts [hunk 6]
```diff
@@ -73,4 +74,19 @@ describe("General Partner Choice Page", () => {
       },
     });
   });
+
+  it("should load the name page with data from api", async () => {
```
Comment (lduranteau) [inline src/presentation/test/integration/registration/general-partner-choice.test.ts:78]: test name 
general partner choice page not name page

File: src/presentation/test/integration/registration/general-partners.test.ts [hunk 7]
```diff
@@ -39,4 +41,19 @@ describe("General Partners Page", () => {
     );
     expect(res.text).toContain(enTranslationText.generalPartnersPage.title);
   });
+
+  it("should load the name page with data from api", async () => {
```
Comment (lduranteau) [inline src/presentation/test/integration/registration/general-partners.test.ts:45]: same as above

File: src/presentation/test/integration/registration/limited-partner-choice.test.ts [hunk 8]
```diff
@@ -90,4 +91,19 @@ describe("Limited Partner Choice Page", () => {
       },
     });
   });
+
+  it("should load the name page with data from api", async () => {
```
Comment (lduranteau) [inline src/presentation/test/integration/registration/limited-partner-choice.test.ts:95]: same as above

File: src/presentation/test/integration/registration/limited-partners.test.ts [hunk 9]
```diff
@@ -39,4 +41,19 @@ describe("Limited Partners Page", () => {
     );
     expect(res.text).toContain(enTranslationText.limitedPartnersPage.title);
   });
+
+  it("should load the name page with data from api", async () => {
```
Comment (lduranteau) [inline src/presentation/test/integration/registration/limited-partners.test.ts:45]: same as above

File: src/presentation/test/integration/registration/general-partner-choice.test.ts [hunk 10]
```diff
@@ -14,6 +14,7 @@ import {
   APPLICATION_CACHE_KEY,
   APPLICATION_CACHE_KEY_PREFIX_REGISTRATION,
 } from "../../../../config/constants";
+import LimitedPartnershipBuilder from "../../builder/LimitedPartnershipBuilder";
 
 describe("General Partner Choice Page", () => {
   beforeEach(() => {
```
Comment (lduranteau) [inline src/presentation/test/integration/registration/general-partner-choice.test.ts:20]: you forgot to reset data in registrationGateway before each testt

File: src/views/limited-partner-choice.njk (hunk 4 above)
Comment (HatiDev) [reply src/views/limited-partner-choice.njk:13]: test added

File: src/views/general-partners.njk (hunk 3 above)
Comment (HatiDev) [reply src/views/general-partners.njk:9]: test added

File: src/views/general-partner-choice.njk (hunk 2 above)
Comment (HatiDev) [reply src/views/general-partner-choice.njk:12]: test added

File: src/views/email.njk (hunk 1 above)
Comment (HatiDev) [reply src/views/email.njk:12]: test added

File: src/presentation/test/integration/registration/general-partner-choice.test.ts (hunk 10 above)
Comment (HatiDev) [reply src/presentation/test/integration/registration/general-partner-choice.test.ts:20]: data reset added

File: src/presentation/test/integration/registration/limited-partners.test.ts (hunk 9 above)
Comment (HatiDev) [reply src/presentation/test/integration/registration/limited-partners.test.ts]: data reset added

File: src/presentation/test/integration/registration/limited-partner-choice.test.ts (hunk 8 above)
Comment (HatiDev) [reply src/presentation/test/integration/registration/limited-partner-choice.test.ts]: data reset added

File: src/presentation/test/integration/registration/general-partners.test.ts (hunk 7 above)
Comment (HatiDev) [reply src/presentation/test/integration/registration/general-partners.test.ts]: data reset added

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

File: src/presentation/test/integration/registration/general-partner-choice.test.ts (hunk 10 above)
Comment (HatiDev) [reply src/presentation/test/integration/registration/general-partner-choice.test.ts:20]: reset data added

File: src/presentation/test/integration/registration/general-partner-choice.test.ts [hunk 11]
```diff
@@ -73,4 +74,19 @@ describe("General Partner Choice Page", () => {
       },
     });
   });
+
+  it("should load the partnership name and name ending with data from api", async () => {
```
Comment (lduranteau) [inline src/presentation/test/integration/registration/general-partner-choice.test.ts:78]: can you update the tests name by something similar to that:
It should contain the proposed name - data from api

Comment (HatiDev) [reply src/presentation/test/integration/registration/general-partner-choice.test.ts]: test name updated

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

File: src/presentation/test/integration/registration/general-partners.test.ts [hunk 12]
```diff
@@ -39,4 +43,19 @@ describe("General Partners Page", () => {
     );
     expect(res.text).toContain(enTranslationText.generalPartnersPage.title);
   });
+
+  it("It should contain the proposed name - data from api", async () => {
```
Comment (lduranteau) [inline src/presentation/test/integration/registration/general-partners.test.ts:47]: the "it" is already there

it("should contain the proposed name - data from api", async () => {

Comment (HatiDev) [reply src/presentation/test/integration/registration/general-partners.test.ts]: removed the 'it'

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Review by lduranteau — APPROVED** [review]

**Pull request merged**